components/camel-mongodb3/src/test/java/org/apache/camel/component/mongodb3/MongoDbOperationsWithConnectionPropertyTest.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.mongodb3;
+
+import java.util.Properties;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.component.properties.PropertiesComponent;
+import org.apache.camel.spring.SpringCamelContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+import com.mongodb.MongoClient;
+
+public class MongoDbOperationsWithConnectionPropertyTest extends MongoDbOperationsTest {
+
+    @Override
+    protected CamelContext createCamelContext() throws Exception {
+        applicationContext = new AnnotationConfigApplicationContext(EmbedMongoConfiguration.class);
```

Comment:
I think this can be simplified by using the camel registry to lookup the bean you are looking for.

Reply:
I replicated the previous test cases that was using the same logic. I could work it around but I should refactor the previous test cases as well...

Reply:
not strictly needed, you can invoke `super.createCamelContext()` here and grab `myDb` and properties component via camel registry

Reply:
The first version was like that but I ended up copying the entire method as I was not aware I could use context to get the property beans! Thanks, I'll try that!